src/solve.jl

```diff
@@ -119,6 +119,31 @@ function solve(system::InitialValueProblem{<:HybridSystem, <:LazySet{N}},
     return solve!(sys_new, copy(options), opC, opD)
 end
 
+
+"""
+    get_necessary_vars(HS::HybridSystem)::Dict{Int,Vector{Int}}
+
+Return all coordinates which appear in any guard or invariant constraint for each location.
+
+### Input
+
+- `HS`  -- hybrid system
+"""
+function get_necessary_vars(HS::HybridSystem)::Dict{Int,Vector{Int}}
+    result = Dict{Int,Vector{Int}}()
```

Comment:
You can initialize the size of the dictionary.

Reply:
> * i'm still not conviced about the name `get_necessary_vars`, what do you think about `constrained_dimensions(::HybridSystem)`?
I agree with it.

> * we should add an example to test this function. we can add it to `solve_hybrid_bouncing_ball.jl` and `solve_hybrid_thermostat.jl`
I thought to have this function private, so we can call it only inside solve. In addition, it will be very 'artificial' test, which will not show real value of this function.